src/transformations.jl

```diff
@@ -270,45 +266,22 @@ function fit(::Type{UnitRangeTransform}, X::AbstractMatrix{<:Real};
         Base.depwarn("fit(t, x) is deprecated: use fit(t, x, dims=2) instead", :fit)
         dims = 2
     end
-    if dims == 1
-        l, tmin, tmax = _compute_extrema(X)
-    elseif dims == 2
-        l, tmin, tmax = _compute_extrema(X')
-    else
-        throw(DomainError(dims, "fit only accept dims to be 1 or 2."))
-    end
-
-    for i = 1:l
-        @inbounds tmax[i] = 1 / (tmax[i] - tmin[i])
-    end
+    dims ∈ [1,2] || throw(DomainError(dims, "fit only accept dims to be 1 or 2."))
+    tmin_tmax = extrema(X; dims=dims)
+    tmin, tmax = (vec(getindex.(tmin_tmax, i)) for i in 1:2)
+    @inbounds @. tmax = 1 / (tmax - tmin)
```

Comment:
`@inbounds` is probably not needed as broadcasting should take care of that.
```suggestion
    @. tmax = 1 / (tmax - tmin)
```